pkg/sql/schemachanger/scplan/internal/rules/current/helpers.go

```diff
@@ -207,6 +207,14 @@ func isConstraint(e scpb.Element) bool {
 	return false
 }
 
+func isCrossDescriptorConstraint(e scpb.Element) bool {
+	switch e.(type) {
+	case *scpb.ForeignKeyConstraint:
+		return true
+	}
+	return false
+}
```

Comment:
yes. 

In general, rule changes we made in this PR are to combat FKs: check constraints are way less "troubling" than FKs in that FKs can form cycles and that caused me a lot of headaches (think: t1->t2, t3->2, dropping t1,t2 cascade). I'm glad that checks don't behave like that, and that the existing rules works for checks even when it references other sequences/types. But I do want it to be principled so I'll add check constraint here.

I also added some logic test where we have cross-descriptor checks and dropping various descriptors in this case works as expected. Note that some of these operations are not even correctly supported in the legacy schema changer due to a known issue (check constraint references sequences by name, rather than by ID), but are correctly handled in the new schema changer.